Save selected subtitle

mathiasvr opened the pull request:
Fixes #701.
The selected subtitle index is stored in the same way as the current time of the playing file. It might be overkill to do this on every render, but I could not see other obvious places to put it, since subtitles can be added/selected in multiple ways.

dcposch commented:
I don't think this works. The subtitles list can come from two different places: either from the torrent, or added by the user. You can drag `.srt` and `.vtt` files onto the player, or select them from the open file dialog after clicking CC.

The subtitles list can change from one time you play a file to the next, so we can't save an index into that list.

dcposch commented:
instead of index, you could save the subtitle path. make sure to distinguish whether it's a relative path (for a file inside the same torrent) or an absolute path (for user supplied subtitles)

mathiasvr commented:
@dcposch You are right, I changed the PR so now it saves the absolute path for the selected subtitle, is that okay?
Also I changed the `addSubtitles` function: if the added file is a duplicate it is still possible to auto-select it. This also makes it possible to re-enable a subtitle by dragging it.

dcposch commented:
LGTM!

mathiasvr commented:
Rebased for #693

dcposch commented:
Thanks!

xnakos commented:
Good job, guys! ;)
